packages/ui/atoms/ScrollAnimation/AbstractScrollAnimation.js

```diff
@@ -0,0 +1,217 @@
+import { Base, withFreezedOptions } from '@studiometa/js-toolkit';
+import { matrix, lerp, map, clamp } from '@studiometa/js-toolkit/utils';
+
+/**
+ * @typedef {{
+ *   x?: number;
+ *   y?: number;
+ *   z?: number;
+ *   scale?: number;
+ *   scaleX?: number;
+ *   scaleY?: number;
+ *   rotate?: number;
+ *   skewX?: number;
+ *   skewY?: number;
+ *   opacity?: number;
+ * }} AnimationOptions
+ */
+
+/**
+ * Get animation configuration defaults.
+ * @returns {AnimationOptions}
+ */
+function getDefaults() {
+  return {
+    x: 0,
+    y: 0,
+    z: 0,
+    scale: 1,
+    scaleX: 1,
+    scaleY: 1,
+    rotate: 0,
+    skewX: 0,
+    skewY: 0,
+    opacity: 1,
+  };
+}
+
+/**
+ * @typedef {AbstractScrollAnimation & {
+ *   $options: {
+ *     playRange: string,
+ *     dampFactor: number,
+ *     dampPrecision: number,
+ *     from: AnimationOptions,
+ *     to: AnimationOptions,
+ *   },
+ * }} ScrollAnimationChildInterface
+ */
+
+/**
+ * AbstractScrollAnimation class.
+ *
+ * @todo Add an option to define start and end offset to allow animation to be mapped
+ * between custom values: start at 0.1 and end at 0.9 for example.
+ *
+ * ```js
+ * offset.x = { start: 0.1, end: 0.5 }
+ *
+ * translateX: lerp(from.x, to.x, map(props.progress.y, offset.x.start, offset.x.end, 0, 1))
+ * ```
+ *
+ * @todo add support for magic values in options to access key values:
+ * - viewport size
+ * - target size
+ * - root element size
+ * - etc.
+ *
+ * ```
+ * data-option-to="{ x: 100%, y: innerWidth / 2 }"
+ * ```
+ */
+export default class AbstractScrollAnimation extends withFreezedOptions(Base) {
+  /**
+   * Config.
+   */
+  static config = {
+    name: 'AbstractScrollAnimation',
+    options: {
+      playRange: {
+        type: Array,
+        default: () => [0, 1],
+      },
+      dampFactor: {
+        type: Number,
+        default: 0.5,
+      },
+      dampPrecision: {
+        type: Number,
+        default: 0.001,
+      },
+      from: {
+        type: Object,
+        default: getDefaults,
+        merge: true,
+      },
+      to: {
+        type: Object,
+        default: getDefaults,
+        merge: true,
+      },
+    },
+  };
+
+  /**
+   * Terminate the class when Web Animation API is not supported.
+   * @returns {void}
+   */
+  mounted() {
+    if (typeof this.target.animate !== 'function') {
```

Comment:
How about adding a test for `prefers-reduced-motion: reduced` media-query for a11y purpose ?
(https://web.dev/prefers-reduced-motion/#remove-motion-on-the-web)

Reply:
I would like to merge this feature quickly, I suggest adding this test on a per-project basis for now.